### PR TITLE
First implementation of Ephemeris

### PR DESCRIPTION
**Pull request by pleroy**
Only Prolong, not Flow, for the time being, and with limited testing (only the Earth-Moon system).

### CODE REVIEW

File: physics/continuous_trajectory_body.hpp [hunk 1]
```diff
@@ -5,8 +5,12 @@
 #include <vector>
 
 #include "physics/continuous_trajectory.hpp"
+#include "testing_utilities/numerics.hpp"
```
Comment (eggrobin) [inline physics/continuous_trajectory_body.hpp]: Perhaps this should be promoted to non-testing numerics.

File: physics/ephemeris_body.hpp [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+  problem.equation = equation_;
+  problem.append_state = std::bind(&Ephemeris::AppendState, this, _1);
+  problem.t_final = t;
+  problem.initial_state = &last_state_;
+
+  // Perform the integration.  Note that we may have to iterate until |t_max()|
+  // actually reaches |t| because the last series may not be fully determined
+  // after the first integration.
+  do {
+    planetary_integrator_.Solve(problem, step_);
```
Comment (eggrobin) [inline physics/ephemeris_body.hpp]: It's not trivial that this isn't stupid. Comment.

File: physics/ephemeris_body.hpp [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    if (body1_is_oblate || body2_is_oblate) {
+      Exponentiation<Length, -2> const one_over_Δq_squared = 1 / Δq_squared;
+      if (body1_is_oblate) {
+        Vector<Acceleration, Frame> const order_2_zonal_acceleration1 =
+            Order2ZonalAcceleration<Frame>(
+                static_cast<OblateBody<Frame> const &>(body1),
+                Δq,
+                one_over_Δq_squared,
+                one_over_Δq_cubed);
+        (*accelerations)[b2] += order_2_zonal_acceleration1;
```
Comment (eggrobin) [inline physics/ephemeris_body.hpp]: This looks like it doesn't respect Newton's third law (see above)...
Investigate.
Later.

File: physics/ephemeris_body.hpp (hunk 2 above)
Comment (pleroy) [inline physics/ephemeris_body.hpp]: Done.

File: physics/ephemeris_body.hpp (hunk 3 above)
Comment (pleroy) [inline physics/ephemeris_body.hpp]: Commented why.

**Pull request merged**